lib/Wikia/src/Util/SamplerProxy.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * A wrapper for a POPO that supports dispatching a specific method to an alternate
+ * object, either as a shadowed execution (i.e. execute both the original method and the
+ * alternate) or as a substitution (i.e. execute the alternate instead of the original).
+ * Both shadowing and substitution may be sampled so that only a specified percentage of
+ * calls involve the alternate and the rest pass through to the original.
+ */
+namespace Wikia\Util;
+
+
+class SamplerProxy {
```

Comment:
This class does two things. It can shadow or dispatch methods. Have you thought of separating those two functionalities?